test/distributed/org/apache/cassandra/distributed/test/ReadRepairEmptyRangeTombstonesTest.java

```diff
@@ -0,0 +1,323 @@
+        cluster.schemaChange(withKeyspace("CREATE TYPE %s.udt (x int, y int)"));
+    }
+
+    @AfterClass
+    public static void teardownCluster()
+    {
+        if (cluster != null)
+            cluster.close();
+    }
+
+    /**
+     * Test range queries asking for an empty range with a tombstone range in one replica.
+     */
+    @Test
+    public void testEmptyRangeQueries()
+    {
+        tester().createTable("CREATE TABLE %s(k int, c int, PRIMARY KEY (k, c)) " +
+                             "WITH CLUSTERING ORDER BY (c %s) AND read_repair='%s'")
+                .executeInternal(1, "DELETE FROM %s USING TIMESTAMP 1 WHERE k=0 AND c>0 AND c<3")
+                .assertRowsDistributed("SELECT * FROM %s WHERE k=0 AND c>1 and c<=1")
+                .assertRowsDistributed("SELECT * FROM %s WHERE k=0 AND c>2 and c<=2")
+                .assertRowsInternal("SELECT * FROM %s")
+                .executeInternal(2, "DELETE FROM %s WHERE k=0 AND c>0 AND c<3");
+    }
+
+    /**
+     * Test range queries asking for an empty range with a tombstone range in one replica, with only static columns.
+     */
+    @Test
+    public void testEmptyRangeQueriesWithStaticRow()
+    {
+        tester().createTable("CREATE TABLE %s(k int, c int, s int static, PRIMARY KEY (k, c)) " +
+                             "WITH CLUSTERING ORDER BY (c %s) AND read_repair='%s'")
+                .executeInternal(1, "DELETE FROM %s USING TIMESTAMP 1 WHERE k=0 AND c>0 AND c<3")
+                .executeInternal("INSERT INTO %s (k, s) VALUES (0, 0)")
+                .assertRowsDistributed("SELECT * FROM %s WHERE k=0 AND c>1 and c<=1")
+                .assertRowsDistributed("SELECT * FROM %s WHERE k=0 AND c>2 and c<=2")
+                .assertRowsInternal("SELECT * FROM %s", row(0, null, 0))
+                .executeInternal(2, "DELETE FROM %s WHERE k=0 AND c>0 AND c<3");
+    }
+
+    /**
+     * Test range queries asking for a not-empty range targeting rows contained in a tombstone range in one replica.
+     */
+    @Test
+    public void testRangeQueriesWithRowsContainedInTombstone()
+    {
+        tester().createTable("CREATE TABLE %s(k int, c int, PRIMARY KEY (k, c)) " +
+                             "WITH CLUSTERING ORDER BY (c %s) AND read_repair='%s'")
+                .executeInternal(1, "DELETE FROM %s USING TIMESTAMP 1 WHERE k=0 AND c>=1 AND c<=5")
+                .executeInternal("INSERT INTO %s (k, c) VALUES (0, 2)")
+                .executeInternal("INSERT INTO %s (k, c) VALUES (0, 3)")
+                .executeInternal("INSERT INTO %s (k, c) VALUES (0, 4)")
+                .assertRowsDistributed("SELECT * FROM %s WHERE k=0 AND c>=2 AND c<=3", row(0, 2), row(0, 3))
+                .assertRowsDistributed("SELECT * FROM %s WHERE k=0 AND c>=3 AND c<=4", row(0, 3), row(0, 4))
+                .assertRowsInternal("SELECT * FROM %s", row(0, 2), row(0, 3), row(0, 4))
+                .executeInternal(2, "DELETE FROM %s WHERE k=0 AND c>=1 AND c<=5");
```

Comment:
@adelapena In the CASSANDRA-15924 tests, we issue this DELETE because it was itself the thing that triggered an error. In this case, we're not issuing an overlapping delete on top of empty ranges, so I wonder if it would make sense to either remove it just verify that the rows are gone w/ `assertRowsInternal()`.

Reply:
I have added a final call to `assertRowsInternal()` after all the final deletions and RR metrics checks. I've also removed the `node` test parameter because it didn't seem useful to produce the scenarios of interest here and it made it difficult to reason about the metrics and final status. I have also added metric checks to `ReadRepairQueryTypesTest`.